community/detectors/geoserver_cve_2024_36401/src/test/java/com/google/tsunami/plugins/detectors/rce/GeoserverCve202436401VulnDetectorTest.java

```diff
@@ -0,0 +1,190 @@
+        .injectMembers(this);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    mockTargetService.shutdown();
+    mockCallbackServer.shutdown();
+  }
+
+  @Test
+  public void detect_whenVulnerable_returnsVulnerability() throws IOException {
+    startMockWebServer();
+    mockCallbackServer.enqueue(PayloadTestHelper.generateMockSuccessfulCallbackResponse());
+
+    DetectionReportList detectionReports =
+        detector.detect(targetInfo, ImmutableList.of(targetNetworkService));
+
+    assertThat(detectionReports.getDetectionReportsList())
+        .containsExactly(
+            DetectionReport.newBuilder()
+                .setTargetInfo(targetInfo)
+                .setNetworkService(targetNetworkService)
+                .setDetectionTimestamp(
+                    Timestamps.fromMillis(Instant.now(fakeUtcClock).toEpochMilli()))
+                .setDetectionStatus(DetectionStatus.VULNERABILITY_VERIFIED)
+                .setVulnerability(
+                    Vulnerability.newBuilder()
+                        .setMainId(
+                            VulnerabilityId.newBuilder()
+                                .setPublisher("TSUNAMI_COMMUNITY")
+                                .setValue("GeoserverCve202436401"))
+                        .setSeverity(Severity.CRITICAL)
+                        .setTitle("Geoserver RCE (CVE-2024-36401)")
+                        .setDescription(
+                            "This detector checks for Geoserver RCE (CVE-2024-36401). "
+                                + "Multiple OGC request parameters allow Remote Code Execution (RCE) "
+                                + "by unauthenticated users through specially crafted input against "
+                                + "a default GeoServer installation due to unsafely evaluating property "
+                                + "names as XPath expressions.")
+                        .setRecommendation(
+                            "Upgrade Geoserver to a patched version. The vulnerability was fixed in versions 2.23.6, 2.24.4, and 2.25.2.")
+                        .addRelatedId(
+                            VulnerabilityId.newBuilder()
+                                .setPublisher("CVE")
+                                .setValue("CVE-2024-36401")))
+                .build());
+    Truth.assertThat(mockTargetService.getRequestCount()).isEqualTo(2);
+    Truth.assertThat(mockCallbackServer.getRequestCount()).isEqualTo(1);
+  }
+
+  @Test
+  public void detect_ifNotVulnerable_doesNotReportVuln() throws IOException {
+    startMockWebServer();
+    mockCallbackServer.enqueue(PayloadTestHelper.generateMockUnsuccessfulCallbackResponse());
+    DetectionReportList detectionReports =
+        detector.detect(targetInfo, ImmutableList.of(targetNetworkService));
+    assertThat(detectionReports.getDetectionReportsList()).isEmpty();
+    Truth.assertThat(mockTargetService.getRequestCount()).isEqualTo(1);
```

Comment:
```suggestion
    Truth.assertThat(mockTargetService.getRequestCount()).isEqualTo(2);
```
You missed this in the last commit